msal/src/test/java/com/microsoft/identity/client/e2e/tests/network/nativeauth/SignUpEmailPasswordTest.kt

```diff
@@ -93,4 +110,148 @@ class SignUpEmailPasswordTest : NativeAuthPublicClientApplicationAbstractTest()
+            assertResult<SignUpResendCodeResult.Success>(resendCodeResult)
+        }
+    }
+
+    /**
+     * Sign up with email + password. User already exists with given email as email-pw account.
+     * (use case 1.1.10, Test case 28)
+     */
+    @Test
+    fun testErrorUserExistAsPassword() {
+        config = getConfig(ConfigType.SIGN_IN_PASSWORD)
+        application = setupPCA(config, defaultChallengeTypes)
+
+        runBlocking { // Running with runBlocking to avoid default 10 second execution timeout.
+            val user = config.email
+            val signUpResult = application.signUp(user)
+            Assert.assertTrue(signUpResult is SignUpError)
+            Assert.assertTrue((signUpResult as SignUpError).isUserAlreadyExists())
+        }
+    }
+
+    /**
+     * Sign up with email + password. User already exists with given email as social account.
+     * (use case 1.1.11, Test case 29)
+     */
+    @Ignore("TODO: Add social account in the tenant.")
+    @Test
+    fun testErrorUserExistAsSocial() {
+        config = getConfig(ConfigType.SIGN_IN_PASSWORD)
+        application = setupPCA(config, defaultChallengeTypes)
+
+        runBlocking { // Running with runBlocking to avoid default 10 second execution timeout.
+            val user = tempEmailApi.generateRandomEmailAddress()
+            val signUpResult = application.signUp(user)
+            assertResult<SignUpResult.CodeRequired>(signUpResult)
+            val codeRequiredState = (signUpResult as SignUpResult.CodeRequired).nextState
+            val resendCodeResult = codeRequiredState.resendCode()
+            assertResult<SignUpResendCodeResult.Success>(resendCodeResult)
+        }
+    }
+
+    /**
+     * Sign up with email + password. Developer makes a request with invalid format email address.
+     * (use case 1.1.12, Test case 30)
+     */
+    @Test
+    fun testErrorInvalidEmailFormat() {
+        config = getConfig(defaultConfigType)
+        application = setupPCA(config, defaultChallengeTypes)
+
+        runBlocking { // Running with runBlocking to avoid default 10 second execution timeout.
+            val user = INVALID_EMAIL
+            val signUpResult = application.signUp(user)
+            Assert.assertTrue(signUpResult is SignUpError)
+            Assert.assertTrue((signUpResult as SignUpError).isInvalidUsername())
+        }
+    }
+
+    /**
+     * Sign up with email + password. Developer makes a request with password that does not match password complexity requirements set on portal.
+     * (use case 1.1.13, Test case 31)
+     */
+    @Test
+    fun testErrorInvalidPasswordFormat() {
+        config = getConfig(defaultConfigType)
+        application = setupPCA(config, defaultChallengeTypes)
+
+        runBlocking { // Running with runBlocking to avoid default 10 second execution timeout.
+            val user = tempEmailApi.generateRandomEmailAddress()
+            val password = INVALID_PASSWORD
```

Comment:
nit: This password does not meet reqs because it doesn't contain numbers or special characters. If the reqs of the tenants change this will start failing as a false positive. I would say a short password such as "pass" is more likely to be invalid because it's too short rather not having numbers/characters

Reply:
Using "pass" as the invalid password.

Reply:
Done.